HelpSource/Guides/UGen-Changelog.schelp

```diff
@@ -0,0 +1,307 @@
+title:: UGen Changelog
+summary:: A summary of UGen changes across SuperCollider versions
+categories:: News, Server>Nodes
+
+This file documents changes affecting UGens in each SuperCollider version,
+starting from version 3.14. 
+
+section:: Version 3.14
+
+Summary of changes:
+list::
+## Decay and Decay2 accept new additional arguments;
+## Added LevelComp pseudo-UGen, providing level compensation logic for
+multichannel mixing and panning
+## GVerb teletype::roomsize:: modulation was fixed;
+## Sweep teletype::rate:: modulation now affects next sample instead of current;
+## ToggleFF doesn't toggle randomly anymore when running at audio rate with a
+control-rate input;
+## EnvGen rounding error when calculating segment duration was fixed;
+## PanAz fixed server crash and numeric blow-up with negative teletype::width::
+
+## Most of the other UGens listed here had their initialization sample
+calculation fixed, affecting their first samples of output;
+::
+
+subsection:: AmpComp
+Fixed initialization sample. No changes in output.
+Previously, calling AmpComp_next when arg rates were anything but all scalar rate caused state vars to be uninitialized and init sample would always be zero.
+Now init sample is always equal to first output sample.
+
+subsection:: Decay
+Fixed initialization sample. First sample of input was incorrectly added twice, fixing it causes a change in the first samples of output:
+code::
+{ Decay.ar(Impulse.ar(0)) }.loadToFloatArray(1e-2, action:_.postcs)
+// 3.13: [1.9998, 1.9995, 1.9992]; 3.14: [1.0, 0.9998, 0.9996]
+// note that in 3.13 values are higher of exactly 1.0, other than being one sample too early
+
+// if input is delayed of at least one sample, no change
+{ Decay.ar(DelayN.ar(Impulse.ar(0), delaytime:1/SampleRate.ir)) }.loadToFloatArray(1e-4, action:_.postln)
+// 3.13: [0.0, 1.0, 0.9998, 0.9996]; 3.14: [0.0, 1.0, 0.9998, 0.9996]
+::
+
+subsection:: Decay2
+Fixed initialization sample. First sample of input was incorrectly added twice, fixing it causes a change in the first samples of output:
+code::
+{ Decay2.ar(Impulse.ar(0), attackTime: 0) }.loadToFloatArray(1e-4, action:_.postln)
+// 3.13: [0.9998, 1.9995, 1.9992]; 3.14: [0.0, 0.9998, 0.9996]
+// note that in 3.13 values are higher of exactly 1.0, other than being one sample too early
+
+// if input is delayed of at least one sample, no change
+{ Decay2.ar(DelayN.ar(Impulse.ar(0), delaytime:1/SampleRate.ir), attackTime: 0) }.loadToFloatArray(1e-4, action:_.postln)
+// 3.13: [0.0, 0.0, 0.9998, 0.9996]; 3.14: [0.0, 0.0, 0.9998, 0.9996]
+::
+
+subsection:: Delay1
+list::
+## Added new argument teletype::x1:: for pre-delay (the value of first output sample,
+defaults to 0.0): teletype::Delay1.ar(in, mul, add, x1: 0.0)::.
```

Comment:
When running at audio rate, x1 defaults to 0.0, i.e. "silence" precedes the delayed input. At control rate, the default value is set to the first sample of `in`, i.e. the first input sample is held during the delay.